### PR TITLE
Software Overview Docs

### PR DESCRIPTION
**Pull request by sugamarora99**

### CODE REVIEW

File: Docs/source/Overview/software-overview.rst [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+
+The BSP interfaces with peripherals and implements interrupt handlers.
+
+Drivers
+=======
+
+Drivers interface with external hardware.
+The software for the BPS has drivers for:
+
+- CAN
```
Comment (manthanand) [inline Docs/source/Overview/software-overview.rst:38]: I would add the :term: syntax here

File: Docs/source/Overview/software-overview.rst [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+- EEPROM
+- LTC6811 (Voltage and Temperature)
+- AS8510 (Amperes)
+
+Real Time Operating System (RTOS)
+=================================
+
+The RTOS uses a scheduler to run multiple threads.
```
Comment (manthanand) [inline Docs/source/Overview/software-overview.rst:46]: Add :term: here for RTOS

File: Docs/source/Overview/software-overview.rst [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+- Pet Watchdog
+- Voltage/Temperature monitor
+- Amperes monitor
+- Battery Balancing
+- Log Information
+- Can Message Consumer
+- Command Line Interface (CLI)
+- Idle  
+
+Descriptions of what each task does can be found in the Tasks section.
```
Comment (manthanand) [inline Docs/source/Overview/software-overview.rst:72]: I would hyperlink to the tasks section for each task above. :rref:

File: Docs/source/Overview/software-overview.rst [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+- :term:`CAN <CAN Bus>`
+- :term:`EEPROM <EEPROM>`
+- LTC6811 (Voltage and Temperature)
+- AS8510 (Amperes)
+
+Real Time Operating System (RTOS)
+=================================
+
+The :term:`RTOS <RTOS>` uses a scheduler to run multiple threads. 
+Running these threads creates "psuedo-concurrency".
```
Comment (ClarkPoon) [inline Docs/source/Overview/software-overview.rst:47]:
```suggestion
Running these threads creates "pseudo-concurrency".
```

File: Docs/source/Overview/software-overview.rst [hunk 5]
```diff
@@ -0,0 +1,74 @@
...
+- :ref:`Fault State <Fault State Task: Manthan Upadhyaya>`
+- :ref:`Critical State <Critical state Task: Manthan Upadhyaya>`
+- :ref:`Pet Watchdog <Pet WatchDog Task: Harshitha Gorla & Clark Poon>`
+- :ref:`Voltage/Temperature monitor <Voltage Temperature Monitor Task: Sijin Woo>`
+- :ref:`Amperes monitor <Amperes Task: Manthan Upadhyaya>`
+- :ref:`Battery Balancing <Battery Balancing Task: Sugam Arora>`
+- :ref:`Log Information <Log Info Task>`
+- :ref:`CAN Message Consumer <CAN Consumer Task: Sugam Arora>`
+- :ref:`Command Line Interface (CLI) <CLI Task: Sugam Arora>`
+- :ref:`Idle <Idle Task>`
```
Comment (ClarkPoon) [inline Docs/source/Overview/software-overview.rst:70]: Can we add priority numbers to these, or make it clear that these are sorted by priority?

File: Docs/source/Overview/software-overview.rst [hunk 6]
```diff
@@ -0,0 +1,74 @@
...
+Real Time Operating System (RTOS)
+=================================
+
+The :term:`RTOS <RTOS>` uses a scheduler to run multiple threads. 
+Running these threads creates "psuedo-concurrency".
+
+The RTOS ensures that safety critical tasks will not be blocked by non-safety
+critical tasks.
+
+We use the Micrium uC/OS-III RTOS.
```
Comment (ClarkPoon) [inline Docs/source/Overview/software-overview.rst:52]: I think it would be nice if we linked to the Micrium Docs here

**Review by ClarkPoon — CHANGES_REQUESTED** [review]

**Review by ClarkPoon — APPROVED** [review]

**Pull request merged**